packages/notebook-extension/src/tool-widgets/metadataEditorFields.tsx

```diff
@@ -0,0 +1,125 @@
+
+    /**
+     * The tracker to the notebook panel.
+     */
+    tracker: INotebookTracker;
+
+    /**
+     * The label of the JSON editor.
+     */
+    label?: string;
+
+    /**
+     * Language translator.
+     */
+    translator?: ITranslator;
+  }
+}
+
+/**
+ * The cell metadata field.
+ *
+ * ## Note
+ * This field does not work as other metadata form fields, as it does not use RJSF to update metadata.
+ * It extends the MetadataEditorTool which updates itself the metadata.
+ * It only renders the node of MetadataEditorTool in a React element instead of displaying a RJSF field.
+ */
+export class CellMetadataField extends NotebookTools.MetadataEditorTool {
+  constructor(options: Private.IOptions) {
+    super(options);
+    this._tracker = options.tracker;
+
+    this.editor.editorHostNode.addEventListener('blur', this.editor, true);
+    this.editor.editorHostNode.addEventListener('click', this.editor, true);
+    this.editor.headerNode.addEventListener('click', this.editor);
+  }
+
+  private _onSourceChanged() {
+    if (this.editor.source) {
+      this._tracker.activeCell?.model.sharedModel.setMetadata(
+        this.editor.source.toJSON()
+      );
+    }
+  }
+
+  render(props: FieldProps): JSX.Element {
+    const cell = this._tracker.activeCell;
+    this.editor.source = cell
+      ? new ObservableJSON({ values: cell.model.metadata as JSONObject })
+      : null;
+    this.editor.source?.changed.connect(this._onSourceChanged, this);
+
+    return (
+      <div className="cell-metadata-editor">
+        <div ref={ref => ref?.appendChild(this.node)}></div>
+      </div>
+    );
+  }
+
+  private _tracker: INotebookTracker;
+}
+
+/**
+ * The notebook metadata field.
+ *
+ * ## Note
+ * This field does not work as other metadata form fields, as it does not use RJSF to update metadata.
+ * It extends the MetadataEditorTool which updates itself the metadata.
+ * It only renders the node of MetadataEditorTool in a React element instead of displaying a RJSF field.
+ */
+export class NotebookMetadataField extends NotebookTools.MetadataEditorTool {
+  constructor(options: Private.IOptions) {
+    super(options);
+    this._tracker = options.tracker;
+
+    this.editor.editorHostNode.addEventListener('blur', this.editor, true);
+    this.editor.editorHostNode.addEventListener('click', this.editor, true);
+    this.editor.headerNode.addEventListener('click', this.editor);
+  }
+
+  private _onSourceChanged() {
+    if (this.editor.source) {
+      this._tracker.currentWidget?.model?.sharedModel.setMetadata(
+        this.editor.source.toJSON()
+      );
+    }
+  }
+
+  render(props: FieldProps): JSX.Element {
+    const notebook = this._tracker.currentWidget;
+    this.editor.source = notebook
+      ? new ObservableJSON({ values: notebook.model?.metadata as JSONObject })
+      : null;
+    this.editor.source?.changed.connect(this._onSourceChanged, this);
+
+    return (
+      <div className="cell-metadata-editor">
```

Comment:
Should all caps global const instead of inline CSS class string be used here?

Please see `shell.ts` for reference:
https://github.com/jupyterlab/jupyterlab/blob/82e1a74b61a41fbe5d65e8542053b749cdfe6ea0/packages/application/src/shell.ts#L39